Dockerfile

```diff
@@ -3,28 +3,34 @@ FROM golang:1.17-bullseye AS builder
 
 WORKDIR /workspace
 
-# Copy the Go Modules manifests
+# Following sections are ordered by expected chance of being changed in daily developer's work:
+# go.mod is expected to change less frequently than the SRO's code.
+# This approach leverages layer caching to speed up rebuilds.
+
 COPY go.mod go.mod
 COPY go.sum go.sum
+RUN ["go",  "mod", "download"]
+
+COPY Makefile* .
```

Comment:
Not sure this works as expected, the build seems to be failing: https://github.com/openshift-psap/special-resource-operator/runs/5536431643?check_suite_focus=true#step:6:56

Reply:
Thanks for bringing it to my attention. This might be a different in podman vs docker.

BTW. From "Build SRO":
> Step 6/32 : COPY Makefile* .
> When using COPY with more than one source file, the destination must be a directory and end with a /

This step ended abruptly, should we catch these?

Reply:
It looks like `minikube image build` returns 0 when it fails, indeed 🙁